common/common/src/main/java/io/helidon/build/common/SubstitutionVariables.java

```diff
@@ -182,9 +182,9 @@ public String resolve(final String value, String replacement) {
      * configured with {@link NotFoundAction#Collapse}.
      */
     public String resolve(final String value, String regex, String replacement) {
-        String resolved = resolve(0, value, value, 0).trim();
+        String resolved = resolve(0, value, value, 0);
         return replacement != null && regex != null
-                ? resolved.replaceAll(regex, replacement)
+                ? resolved.trim().replaceAll(regex, replacement)
```

Comment:
This does not feel right. The contract of this code is to substitute variables as-is. Using `trim()` here is very likely to have unintended side effects.

The real issue here is about "external" values supplied to the archetype engine. We should probably
sanitize values known values (GAV, package, etc.).

Reply:
Right, I was focus on the `user.name` issue only.